blaze/expr/collections.py

```diff
@@ -418,6 +418,7 @@ def on_right(self):
         return on_right
 
     @property
+    @memoize
     def schema(self):
```

Comment:
> could we just make schema a descriptor that swaps itself out on the instance

@llllllllll can you give an example of this? i don't quite understand what you mean here.